packages/react-hooks/src/CredentialProvider.tsx

```diff
@@ -1,30 +1,35 @@
-import type { RecordsState } from './recordUtils'
-import type { Agent, CredentialState } from '@aries-framework/core'
+import type { CombinedRecordsState, RecordsState } from './recordUtils'
+import type { Agent, CredentialState, GetFormatDataReturn, IndyCredentialFormat } from '@aries-framework/core'
 import type { PropsWithChildren } from 'react'
 
 import { CredentialExchangeRecord } from '@aries-framework/core'
 import { useState, createContext, useContext, useEffect, useMemo } from 'react'
 import * as React from 'react'
 
 import {
+  addCombinedRecord,
+  removeCombinedRecord,
+  updateCombinedRecord,
   recordsRemovedByType,
   recordsUpdatedByType,
   recordsAddedByType,
-  removeRecord,
-  updateRecord,
-  addRecord,
 } from './recordUtils'
 
-const CredentialContext = createContext<RecordsState<CredentialExchangeRecord> | undefined>(undefined)
+const CredentialContext = createContext<CombinedRecordsState<CredentialExchangeRecord> | undefined>(undefined)
 
-export const useCredentials = () => {
+export const useCombinedCredentials = () => {
```

Comment:
Maybe useCredentialsWithFormatData()? Or do you think that is too long? 

combined doesn't really give any context on what is combined